AmplifyPlugins/Core/AWSPluginsCore/Model/Support/Model+GraphQL.swift

```diff
@@ -17,7 +17,7 @@ extension Model {
     /// used as the `input` of GraphQL related operations.
     func graphQLInput(_ modelSchema: ModelSchema) -> GraphQLInput {
         var input: GraphQLInput = [:]
-        modelSchema.fields.forEach {
+        modelSchema.fields.filter { !$0.value.isReadOnly }.forEach {
```

Comment:
could we move this to line 23? It would keep move the logic out into the body of the for loop, and make the placement of a comment easier

```swift
// When the field is a read-only, do not add it to the GraphQL input object
            if modelField.isReadOnly {
                return
            }
```


I'm not sure if it is better to put it in the ModelDecorator as how the code is right now. In ModelDecorator, it's getting the GraphQL Input from the model, and then modifying it based on the schema's auth rules, such as removing "ownerFields". Since the read-only field is on the field, and not the schema level, i think it's probably better to keep the logic in this computed property with the rest of the field related logic

Reply:
thanks, applied that suggestion and renamed `graphQLInput` to `graphQLInputForMutation`